src/Compilers/CSharp/Test/Emit2/Semantics/LockTests.cs

```diff
@@ -1441,6 +1441,118 @@ public void LangVersion()
             verify: Verification.FailsILVerify).VerifyDiagnostics();
     }
 
+    [Fact]
+    public void OperationTree()
+    {
+        var source = """
+            using System;
+            using System.Threading;
+
+            Lock l = new Lock();
+            /*<bind>*/lock (l)
+            {
+            }/*</bind>*/
+            """;
+
+        var expectedOperationTree = """
+            ILockOperation (OperationKind.Lock, Type: null) (Syntax: 'lock (l) ... }')
+              Expression:
+                ILocalReferenceOperation: l (OperationKind.LocalReference, Type: System.Threading.Lock) (Syntax: 'l')
+              Body:
+                IBlockOperation (0 statements) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+            """;
+
+        var expectedDiagnostics = DiagnosticDescription.None;
+
+        VerifyOperationTreeAndDiagnosticsForTest<LockStatementSyntax>([source, LockTypeDefinition], expectedOperationTree, expectedDiagnostics);
+    }
+
+    [Fact]
+    public void FlowGraph()
+    {
+        var source = """
+            using System;
+            using System.Threading;
+
+            class C
+            {
+                void M()
+                /*<bind>*/{
+                    Lock l = new Lock();
+                    lock (l)
```

Comment:
>l [](http://example.com/codeflow?start=26&length=1)

Please also test scenario with a control flow at this position. For example, a conditional expression, a coalesce expression, etc.